sdk/v1/client.go

```diff
@@ -30,28 +29,36 @@ type Client struct {
 	serviceToken string
 }
 
-// NewWithSetTimeoutAndMaxRetry creates a new SearchReindexApi Client, with a configurable timeout and maximum number of retries
-func NewClientWithClienter(serviceToken, searchReindexURL string, clienter dphttp.Clienter) *Client {
-	hcClient := healthcheck.NewClientWithClienter(service, searchReindexURL, clienter)
-
+// New creates a new instance of Client with a given search reindex api url
+func New(searchReindexAPIURL, serviceToken string) *Client {
 	return &Client{
 		apiVersion:   apiVersion,
-		hcCli:        hcClient,
+		hcCli:        healthcheck.NewClient(service, searchReindexAPIURL),
 		serviceToken: serviceToken,
 	}
 }
 
-// NewWithHealthClient creates a new instance of Client,
-// reusing the URL and Clienter from the provided health check client.
-func NewClientWithHealthcheck(serviceToken string, hcCli *healthcheck.Client) *Client {
+// NewWithHealthClient creates a new instance of SearchReindexAPI Client,
+// reusing the URL and Clienter from the provided healthcheck client
+func NewWithHealthClient(serviceToken string, hcCli *healthcheck.Client) *Client {
 	return &Client{
```

Comment:
(minor): I think here you would have to check if the hcCli is nil and if it is nil probably return a nil client and an error ?